src/omnicore/tx.cpp

```diff
@@ -237,10 +316,20 @@ bool CMPTransaction::interpret_SendToOwners()
 /** Tx 4 */
 bool CMPTransaction::interpret_SendAll()
 {
-    if (pkt_size < 5) {
-        return false;
+    if (encodingClass == OMNI_CLASS_D) {
+        int i = 0;
+
+        std::vector<uint8_t> vecVersionBytes = GetNextVarIntBytes(i);
+        std::vector<uint8_t> vecTypeBytes = GetNextVarIntBytes(i);
+
+        memcpy(&ecosystem, &pkt[i], 1);
```

Comment:
Yep, you're right - fixed on each of the occurrences you noted (plus two more for early bird and percentage in `interpret_CreatePropertyVariable()`).